File: Cora/Objective-C/Managers/ColorManager.m
#import "ColorManager.h"


@implementation ColorManager


+ (ColorManager *)sharedInstance {

	static ColorManager *sharedInstance = nil;
	static dispatch_once_t onceToken;

	dispatch_once(&onceToken, ^{ sharedInstance = [self new]; });

	return sharedInstance;

}


- (void)loadInitialAccentColor {

	self.accentColor = UIColor.greenColor;

	NSData *encodedData = [NSKeyedArchiver archivedDataWithRootObject: self.accentColor requiringSecureCoding:NO error:nil];
	NSDictionary *defaultAccentColor = [NSDictionary dictionaryWithObject: encodedData forKey:@"accentColor"];
	[NSUserDefaults.standardUserDefaults registerDefaults:defaultAccentColor];

}


@end

File: Released/Cora/UIKit/Managers/ColorManager.swift
import UIKit


final class ColorManager {

	static let sharedInstance = ColorManager()
	var accentColor = UIColor.green

	private init() {}

	func loadInitialAccentColor() {
		do {
			let encodedData = try NSKeyedArchiver.archivedData(withRootObject: accentColor, requiringSecureCoding: false)
			let defaultAccentColor = ["accentColor": encodedData]
			UserDefaults.standard.register(defaults: defaultAccentColor)
		}
		catch { return }
	}

}
